signac/contrib/project.py

```diff
@@ -1531,6 +1531,49 @@ def get_project(cls, root=None, search=True):
                 "Unable to determine project id for path '{}'.".format(os.path.abspath(root)))
         return cls(config=config)
 
+    @classmethod
+    def get_job(cls, root=None):
+        """Find a Job in or above the current working directory (or
+        provided path).
+
+        :param root: The job root directory.
+            If no root directory is given, the current working directory is
+            assumed to be the job directory.
+        :type root: str
+        :returns: The job handle.
+        :raises LookupError: If this job cannot be found."""
+        if root is None:
+            root = os.getcwd()
+        root = os.path.abspath(root)
+
+        # Ensure the root path exists, which is not guaranteed by the regex match
+        if not os.path.exists(root):
+            raise LookupError("Path does not exist: '{}'.".format(root))
+
+        # Find the last match instance of a job id
+        results = list(re.finditer(JOB_ID_REGEX, root))
+        if len(results) == 0:
+            raise LookupError("Could not find a job id in path '{}'.".format(root))
+        match = results[-1]
+        job_id = match.group(0)
+        job_root = root[:match.end()]
+
+        # Find a project *above* the root directory (if the provided root
+        # contains a project, we must search from its parent)
+        project = cls.get_project(os.path.join(job_root, os.pardir))
+
+        # Search for the matched job id in the found project
+        try:
+            job = project.open_job(id=job_id)
+        except KeyError:
+            raise
+        else:
+            if job in project:
+                return job
+            else:
+                raise LookupError("Unable to find job id {} in project '{}'.".format(
```

Comment:
I think we can just rely on the exception raised by the `open_job()` function in line 1567, can we not?